sympy/solvers/diophantine/diophantine.py

```diff
@@ -3614,29 +3633,38 @@ def prime_as_sum_of_two_squares(p):
     =========
 
     .. [1] Representing a number as a sum of four squares, [online],
-        Available: https://schorn.ch/lagrange.html
+           Available: https://schorn.ch/lagrange.html
 
     See Also
     ========
-    sum_of_squares()
+
+    sum_of_squares
+
     """
-    if not p % 4 == 1:
+    p = as_int(p)
+    if p % 4 != 1:
         return
+    if not isprime(p):
+        raise ValueError("p should be a prime number")
 
     if p % 8 == 5:
+        # Legendre symbol (2/p) == -1 if p % 8 in [3, 5]
         b = 2
-    else:
+    elif p % 12 == 5:
+        # Legendre symbol (3/p) == -1 if p % 12 in [5, 7]
         b = 3
-
-        while pow(b, (p - 1) // 2, p) == 1:
+    elif p % 5 in [2, 3]:
```

Comment:
OK, I'll bite: why do we explicitly test `in [2,3]` but not do so above, e.g. like `p%8 in [3,5]`?

Reply:
I see - being 1 from 4 means the most you can be from multiples of 4 greater than that is 5.